### PR TITLE
2022 03 19 glib update

### LINKED ISSUE #23647: [glib] update to 2.70.5
Library name: glib

New version number: 2.70.5

Other information that may be useful (release notes, etc...)
https://github.com/GNOME/glib/commits/glib-2-70

This same-abi-branch release fixes a quite considerable number of invalid memory accesses.

I am aware we also have #22842 which has stalled for 2 months now. The 2.70 branch update is rather "minor", but doesn't address the static linking issue as the upstream fix was so far introduced only into the 2.71 and newer branches

### LINKED ISSUE #23649: [glibmm] update to 2.70
Library name: glibmm

New version number: 2.70

Other information that may be useful (release notes, etc...)
https://github.com/GNOME/glibmm/commits/glibmm-2-70

### LINKED ISSUE #23650: [glibmm] support msvc2022
****

**Is your feature request related to a problem? Please describe.**
When compiling glibmm using msvc2022, the customized toolchain selection done in the port's makefile will use the previous generation toolchain from the windows sdk.

**Proposed solution**
There's a patch upstream that solves the issue but it didn't make it into the last release. Apply the patch on our side

**Describe alternatives you've considered**
None. 

**Additional context**
export and binarycache metadata will indicate the usage of a v143  compiler toolchain from an msvc2022 installation, while the one that has actually been used is the v142 one from the windows sdk or vc-buildtools, resulting in the package's metadata being inconsistent to the package's content.

### PR DESCRIPTION
**Pull request by SchaichAlonso**
**Describe the pull request**
Update glib and it's C++ wrapper (glibmm).

- #### What does your PR fix?  
  Fixes #23647 , #23649 and #23650 .

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
(https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  any linux and non-static windows, CI baseline was not updated

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/glibmm/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "glibmm",
-  "version": "2.68.1",
-  "port-version": 2,
+  "version": "2.70.0",
+  "port-version": 1,
```
Comment (Thomas1664) [inline ports/glibmm/vcpkg.json:4]:
```suggestion
  "version": "2.70.0",
```

File: versions/g-/glibmm.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "e1dc9b8da6a1e348d6d8aab8b969fdbc4c84d74b",
+      "version": "2.70.0",
+      "port-version": 1
+    },
```
Comment (Thomas1664) [inline versions/g-/glibmm.json:7]:
```suggestion
```
Please remove this and run `vcpkg x-add-version glibmm --overwrite-version` after you committed all changes.

File: versions/baseline.json [hunk 3]
```diff
@@ -2509,12 +2509,12 @@
...
-      "baseline": "2.70.1",
-      "port-version": 2
+      "baseline": "2.70.5",
+      "port-version": 0
     },
     "glibmm": {
-      "baseline": "2.68.1",
-      "port-version": 2
+      "baseline": "2.70.0",
+      "port-version": 1
```
Comment (Thomas1664) [inline versions/baseline.json:2517]:
```suggestion
      "port-version": 0
```

**Review by Thomas1664 — CHANGES_REQUESTED** [review]

File: versions/baseline.json (hunk 3 above)
Comment (SchaichAlonso) [reply versions/baseline.json:2517]: done and re-pushed

**Comment by Thomas1664** [comment]
Please run `vcpkg x-add-version glibmm --overwrite-version`. Because you modified `ports/glibmm/vcpkg.json` the old commit hash isn't valid anymore.

**Review by Thomas1664 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**